src/System.Management.Automation/engine/parser/TypeResolver.cs

```diff
@@ -109,6 +117,10 @@ private static Type LookForTypeInAssemblies(TypeName typeName, IEnumerable<Assem
                             }
                         }
                     }
+                    else
+                    {
+                        searchedAssemblies.Add(assembly.FullName);
```

Comment:
Why is this in the `else` block - shouldn't you add unconditionally? Or at least if `targetType` is null?

It looks like you might still have a bug if the name is the alternateNameToFind, e.g. `Cmdlet` or `CmdletAttribute` - we search for both in an attribute context.